File: Sean的leetcode/python/Summary Ranges.py
###
# Array
# Time Complexity: O(n)
# Space Complexity: O(n)
###
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        if not nums:
            return []
        re = []
        pre = nums[0]
        for i, n in enumerate(nums):
            if i == len(nums) - 1 or nums[i+1] != n + 1:
                if pre == n:
                    re.append(str(pre))
                else:
                    re.append(str(pre) + "->" + str(n))
                if i != len(nums) - 1:
                    pre = nums[i+1]
        return re
                
        
        

File: MyLeetCode/Leetcode/Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> res;
        int size = nums.size();
        if (size == 0) return res;
        int start = nums[0];
        for (int i = 0; i < size; i++) {
            if (i == size - 1 || nums[i+1] != nums[i] + 1) {
                if (start == nums[i]) {
                    res.push_back(to_string(start));
                }
                else {
                    res.push_back(to_string(start) + "->" + to_string(nums[i]));
                }
                if (i != size - 1) {
                    start = nums[i+1];
                }
            }
        }
        return res;
    }
};